### DISCUSSION TITLE
[question] How to design the IoC container in multi type services?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [inversify/InversifyJS#1580] [Q&A] [answered]
I have a project as follows:

```
src/main/database/ // classes for database store
  services/
  interfaces/
src/main/communicate/ // classes for communication
  services/
  interfaces/
src/main/ui/ // classes for UI
  services/
  interfaces/
...
src/other/usage
```
I am currently designing to use an IoC container to decouple the definitions under `src/main/` for `src/other/usage`. 
I can create an IoC container for each directory under the `src/main/` directory. That is to say, in my project, there are different containers for different service types. Then when used in `src/other/usage`, multiple containers need to be imported.

Or should only one container be created, for example: `src/ioc.ts` 

```
src/main/database/ 
src/main/communicate/ 
src/main/ui/
```

then register all services to this one IoC container?

**Comment by notaphplover** [comment, marked as answer]
I would personally use [ContainerModule](https://github.com/inversify/InversifyJS/blob/master/wiki/container_modules.md) to distribute the responsibility of binding your services.
